test/ad/utils.jl

```diff
@@ -78,27 +78,39 @@ function test_ad(f, x, broken=(); rtol=1e-6, atol=1e-6)
         end
     end
 
-    if (AD == "All" || AD == "Tapir") && VERSION >= v"1.10"
-        rule = Tapir.build_rrule(f, x; safety_on=false)
-        if :tapir in broken
-            @test_broken(
-                isapprox(
-                    Tapir.value_and_gradient!!(rule, f, x)[2][2],
-                    finitediff;
-                    rtol=rtol,
-                    atol=atol,
-                )
-            )
-        else
-            @test(
-                isapprox(
-                    Tapir.value_and_gradient!!(rule, f, x)[2][2],
-                    finitediff;
-                    rtol=rtol,
-                    atol=atol,
-                )
-            )
+    if (AD == "All" || AD == "Mooncake") && VERSION >= v"1.10"
+        try
+            Mooncake.build_rrule(f, x)
+        catch exc
+            # TODO(penelopeysm):
+            # @test_throws AssertionError (expr...) doesn't work, unclear why
+            @test exc isa AssertionError
         end
```

Comment:
I don't know why `@test_throws` doesn't work here, it's completely inexplicable.

However, I also didn't want to spend another 2 hours debugging this, so I'm just reverting to catching the expression and testing it.

Reply:
BTW, we could simply exclude Mooncake from testing entirely, but I prefer having this as it will let us know when the bug above is fixed.